File: test/fixtures/php/Person.php
<?php

/**
 * A person with a name and age.
 */
class Person {
    private string $name;
    private int $age;

    public function __construct(string $name, int $age) {
        $this->name = $name;
        $this->age = $age;
    }

    /** Returns a greeting string. */
    public function greet(): string {
        return "Hello, " . $this->name;
    }

    public static function add(int $x, int $y): int {
        return $x + $y;
    }
}

$p = new Person("Alice", 30);
echo $p->greet();
echo Person::add(1, 2);

File: test/fixtures/lua/person.lua
--- A person with a name and age.
---@class Person
---@field name string
---@field age integer
local Person = {}
Person.__index = Person

---@param name string
---@param age integer
---@return Person
function Person.new(name, age)
    local self = setmetatable({}, Person)
    self.name = name
    self.age = age
    return self
end

---@return string
function Person:greet()
    return "Hello, " .. self.name
end

  
return Person
